src/Microsoft.Health.Fhir.Core/Features/Context/FhirRequestContext.cs

```diff
@@ -62,6 +62,6 @@ public class FhirRequestContext : IFhirRequestContext
 
         public IDictionary<string, StringValues> ResponseHeaders { get; }
 
-        public string ResourceType { get; }
+        public string ResourceType { get; set; }
```

Comment:
I think for RouteName and ResourceType, we should consider throw an example if the value has not be set to avoid weird bugs down the road where depending when it's access, it might return data or null.

Reply:
I tried to do that, but endpoint like health/check would fail in that case, where it doesn't go through the MVC pipeline. I discussed it with John and we decided that it should be OK to leave it without throwing an exception, and having tests that ensure it's set for the endpoints we expect it to be set.